solidity/contracts/contracts/HashingTest.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.0;
+
+contract HashingTest {
+    bytes32 public lastCheckpoint;
+    address[] public state_validators;
+    uint256[] public state_powers;
+    uint256 public state_nonce;
+
+    function IterativeHash(
+        address[] memory _validators,
+        uint256[] memory _powers,
+        uint256 _valsetNonce,
+        bytes32 _peggyId
+    ) public {
+        // bytes32 encoding of the string "checkpoint"
+        bytes32 methodName = 0x636865636b706f696e7400000000000000000000000000000000000000000000;
```

Comment:
You can move this to the contract scope and declare it constant like:

```    bytes32 constant methodName = 0x636865636b706f696e7400000000000000000000000000000000000000000000;   
```

Instead of copying it into every function.

Reply:
Fixed in dc67c26.